mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java

```diff
@@ -146,25 +143,18 @@ private static UidCriterion findConjugatedUidCriterion(List<Criterion> crits) {
         return ImmutableList.copyOf(new MessageSearches(hitSet.iterator(), query, textExtractor).iterator());
     }
 
-    private boolean isMatchingUser(MailboxSession session, Mailbox mailbox) {
-        return mailbox.getUser().equals(session.getUser().getUserName());
-    }
-
     @Override
-    public List<MessageId> search(MailboxSession session, final MultimailboxesSearchQuery searchQuery, long limit) throws MailboxException {
-        List<Mailbox> allUserMailboxes = mailboxMapperFactory.getMailboxMapper(session)
-                .findMailboxWithPathLike(new MailboxPath(session.getPersonalSpace(), session.getUser().getUserName(), WILDCARD));
-        Stream<Mailbox> filteredMailboxes = allUserMailboxes.stream()
-            .filter(mailbox -> !searchQuery.getNotInMailboxes().contains(mailbox.getMailboxId()));
-
-        if (searchQuery.getInMailboxes().isEmpty()) {
-            return getAsMessageIds(searchResults(session, filteredMailboxes, searchQuery.getSearchQuery()), limit);
+    public List<MessageId> search(MailboxSession session, final Collection<MailboxId> mailboxIds, SearchQuery searchQuery, long limit) throws MailboxException {
+        MailboxMapper mailboxManager = mailboxMapperFactory.getMailboxMapper(session);
+        ImmutableList.Builder<Mailbox> builder = ImmutableList.builder();
+
+        for(MailboxId mailboxId : mailboxIds) {
+            builder.add(mailboxManager.findMailboxById(mailboxId));
         }
-        List<Mailbox> queriedMailboxes = filteredMailboxes
-            .filter(mailbox -> searchQuery.getInMailboxes().contains(mailbox.getMailboxId()))
-            .collect(Guavate.toImmutableList());
 
-        return getAsMessageIds(searchResults(session, queriedMailboxes.stream(), searchQuery.getSearchQuery()), limit);
+        Stream<Mailbox> filteredMailboxes = builder.build().stream();

```

Comment:
mailboxIds.stream().map(mailboxManager::findMailboxById)

Reply:
I can't use lambda here because of checked exception!